### ISSUE TITLE
[Bug]: Invalid location information for regexp:RegExp symbol

### ISSUE THREAD

**Issue by malinthar** [ballerina-platform/ballerina-lang#39478]
### Description

Consider the regexp:RegExp symbol depicted below

```ballerina
import ballerina/lang.regexp;

public function main() {
    regexp:RegExp reg = re ``;
}

```

The location information of the RegExp symbol us invalid as depicted below

![image](https://user-images.githubusercontent.com/35211477/216231585-09702743-001d-4c1a-92d6-6513903a6552.png)

![image](https://user-images.githubusercontent.com/35211477/216231615-ae554cc5-e8c0-4f90-8a2a-c99c8584a613.png)

### Related area

-> Compilation

**Comment by malinthar** [comment]
When investigated found that the tSymbol of the BTypeReferenceType contains the `bulitIn` location instead of the user defined location. Seems to be an issue in the BIRTypeWriter

**Comment by pcnfernando** [comment]
https://github.com/ballerina-platform/ballerina-lang/pull/39509 should fix the issue. The issue was that regexp builtin-subtype's typeSymbol was not updated to point to regexp pkg. Because of this, at BIRGen the type was updated to typereference type at https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java#L387

**Comment by pcnfernando** [comment]
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/39509. @malinthar please check

**Comment by malinthar** [comment]
> Fixed with #39509. @malinthar please check

Closing as fixed
